src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c

```diff
@@ -21,19 +31,178 @@ const ActorInit Obj_Comb_InitVars = {
     (ActorFunc)ObjComb_Update,
     (ActorFunc)ObjComb_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Comb/func_80B91B60.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Comb/func_80B91E4C.s")
+UNK_TYPE D_80B922E0[] = {
+    0x00000000, 0x00000000, 0x00000000, 0x4001FFFE, 0x00000000, 0x00010100, 0x00000000, 0x00000000, 0x000F0064,
+};
+
+UNK_TYPE D_80B92304[] = {
+    0x0A000909,
+    0x20000000,
+    0x00000001,
+    &D_80B922E0,
+};
+
+InitChainEntry initChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F4, 1100, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 900, ICHAIN_STOP),
+};
+
+extern UNK_TYPE D_050095B0;
+extern UNK_TYPE D_05009940;
+
+void ObjComb_Break(ObjComb* this, GlobalContext* globalCtx) {
+    Vec3f pos1;
+    Vec3f posSum;
+    Vec3f pos2;
+    Gfx** dlist = &D_05009940;
+    s16 scale;
+    s16 phi_s2 = 0;
+    s16 gravityInfluence;
+    char u0;
+    char rotSpeed;
+    f32 rand1;
+    f32 rand2;
+    s32 i;
+
+    for (i = 0; i < 31; i++) {
+        phi_s2 += 20000;
+        rand1 = Math_Rand_ZeroOne() * 10.0f;
+
+        pos1.x = Math_Sins(phi_s2) * rand1;
+        pos1.y = (i - 15) * 0.7f;
+        pos1.z = Math_Coss(phi_s2) * rand1;
+
+        Math_Vec3f_Sum(&pos1, &this->actor.posRot.pos, &posSum);
+
+        pos2.x = (Math_Rand_ZeroOne() - 0.5f) + pos1.x * 0.5f;
+        pos2.y = (Math_Rand_ZeroOne() - 0.5f) + pos1.y * 0.6f;
+        pos2.z = (Math_Rand_ZeroOne() - 0.5f) + pos1.z * 0.5f;
+
+        scale = Math_Rand_ZeroOne() * 72.0f + 25.0f;
+        if (scale < 40) {
+            gravityInfluence = -200;
+            rotSpeed = 40;
+        } else if (scale < 70) {
+            gravityInfluence = -280;
+            rotSpeed = 30;
+        } else {
+            gravityInfluence = -340;
+            rotSpeed = 20;
+        }
+
+        rand2 = Math_Rand_ZeroOne();
+        if (rand2 < 0.1f) {
+            u0 = 96;
+        } else if (rand2 < 0.8f) {
+            u0 = 64;
+        } else {
+            u0 = 32;
+        }
+        Effect_SpawnFragment(globalCtx, &posSum, &pos2, &posSum, gravityInfluence, u0, rotSpeed, 4, 0, scale, 0, 0, 80,
+                             -1, 2, dlist);
+    }
+
+    posSum.x = this->actor.posRot.pos.x;
+    posSum.y = this->actor.posRot.pos.y - 10.0f;
+    posSum.z = this->actor.posRot.pos.z;
+    func_80033480(globalCtx, &posSum, 40.0f, 6, 70, 60, 1);
+}
+
+void func_80B91E4C(ObjComb* this, GlobalContext* globalCtx) {
+    s16 params = this->actor.params & 0x1F;
+
+    if ((params > 0) || (params < 0x1A)) {
+        if (params == 6) {
+            if (Flags_GetCollectible(globalCtx, (this->actor.params >> 8) & 0x3F) != 0) {
```

Comment:
Since this is a flag we usually omit the != 0 unless it doesn't match without it.  
```suggestion
            if (Flags_GetCollectible(globalCtx, (this->actor.params >> 8) & 0x3F) {
```